### PR TITLE
Add fields google tag manager

### PR DESCRIPTION
**Pull request by cholletNovius**

### CODE REVIEW

**Comment by cholletNovius** [comment]
google seo tools

File: bootstrap.php [hunk 1]
```diff
@@ -23,45 +22,66 @@
...
+    if (!empty($fullScript) || !empty($fullScriptTagManagerHead)) {
+        preg_match("/<\/head>/", $html, $matches);
+        if (!empty($matches) && isset($matches[0])) {
+            $html = str_replace($matches[0], "\n".$fullScript.$fullScriptTagManagerHead."\n".$matches[0], $html);
+        }
+    }
+
+    //after body
+    $fullScriptTagManagerBody = \Bru\Google\Seo\Tools\Tools_Google_Seo::getTagmanagerTrackingScriptBody();
+    if(empty($fullScriptTagManagerBody)) return false;
```
Comment (shaoshiva) [inline bootstrap.php]: Code style !

File: classes/tools/google/seo.php [hunk 2]
```diff
@@ -48,80 +47,150 @@ public static function getAnalyticsTrackingScript() {
...
+
+        if ($config['tagmanager_full_script_body'] != '') {
+            if (\Str::starts_with($config['tagmanager_full_script_body'], '<script')) {
+                $fullScript = $config['tagmanager_full_script_body'];
+            } else {
+                $fullScript = '<script type="text/javascript">'.$config['tagmanager_full_script_body'].'</script>';
+            }
+        } else {
+            $tag = \Arr::get($config, 'google_tagmanager_tag');
+            $fullScript = \View::forge('bru_google_seo_tools::js_tagmanager_body', compact('tag'));
```
Comment (shaoshiva) [inline classes/tools/google/seo.php]: Third parameter to `false` on `View::forge()` to prevent auto-escape ?

File: classes/tools/google/seo.php [hunk 3]
```diff
@@ -48,80 +47,150 @@ public static function getAnalyticsTrackingScript() {
...
+            if (\Str::starts_with($config['tagmanager_full_script_head'], '<script')) {
+                $fullScript = $config['tagmanager_full_script_head'];
+            } else {
+                $fullScript = '<script type="text/javascript">'.$config['tagmanager_full_script_head'].'</script>';
+            }
         } else {
-            self::_doNotTrack($full_script);
-            return (string)$full_script;
+            $tag = \Arr::get($config, 'google_tagmanager_tag');
+            $fullScript = \View::forge('bru_google_seo_tools::js_tagmanager_head', compact('tag'));
```
Comment (shaoshiva) [inline classes/tools/google/seo.php]: Same as above, third parameter to false on View::forge() to prevent auto-escape ?

**Review by shaoshiva — APPROVED** [review]
:+1:

**Review by jguyomard — APPROVED** [review]

File: config/controller/admin/config.config.php [hunk 4]
```diff
@@ -55,7 +55,7 @@
                     ),
                 ),
             ),
-            99 => array(
+            2 => array(
```
Comment (DOMEYD) [inline config/controller/admin/config.config.php]: Maybe a reason to use 99 ? Another website may overwrite 2 or 3 ?

Comment (cholletNovius) [reply config/controller/admin/config.config.php]: I yes your maybe true ! : but why the before key (https://github.com/novius/bru_google_seo_tools/pull/4/files#diff-cba3c1bfda1f498c5955313ee026df53R10) => `1` will not be overwrite also by another website ?

**Pull request merged**